src/Reflection/InitializerExprTypeResolver.php

```diff
@@ -1616,63 +1626,93 @@ private function integerRangeMath(Type $range, BinaryOp $node, Type $operand): T
 				}
 			}
 		} elseif ($node instanceof Expr\BinaryOp\Mul) {
-			if ($operand instanceof ConstantIntegerType) {
-				/** @var int|float|null $min */
-				$min = $rangeMin !== null ? $rangeMin * $operand->getValue() : null;
+			$min1 = ($rangeMin ?? -INF) * ($operandMin ?? -INF);
```

Comment:
Multiplication was wrong for thing like
```
int<-1, 2> * int<-1, 2>
```
we shouldn't always multiply both min and both max.
Sometimes it's one min multiplied by the other max.

Multiplication and using min/max function wasn't easy with the `null` value. The easiest solution I found is to use `-INF` and `INF` instead of null.